### PR TITLE
i18n proof-of-concept

### PR DESCRIPTION
**Pull request by plata**
requires PlayOnLinux/POL-POM-5#872

### CODE REVIEW

**Comment by plata** [comment]
As the translations are stored in json, you cannot use the real string as key (e.g. no dots allowed). Also, I would like to replace `i18next.t` by `tr`. Anyone knows how to do this?

**Comment by madoar** [comment]
Does `val tr = i18next.t;` work?

**Comment by qparis** [comment]
I think translation context is typically a thing that should be injected from Java

**Comment by qparis** [comment]
Thinking about it, I don't this we should use another i18 framework written in JavaScript.
Why not using java's tr directly?

**Comment by plata** [comment]
@madoar I've tried that already. Doesn't work for some reason but I don't understand why.

**Comment by plata** [comment]
@qparis I've tried to explain it in #49, but let me try again: gettext-commons creates a `Messages_lang.properties` or `Messages_lang.class` (e.g. `Messages_de.properties`) which contains the translated Strings. This is then somehow included in the jar. If we want to use `tr` in JS as well, all Strings from JS have to be in those `.properties`/`.class` files. While we could control that for our own repository, it would become practically impossible for any one else to translate their own repositories. With my proposed solution we read the translation files from the repository at runtime which solves this issue.

If you know a way to inject the translation files from the repositories into gettext-commons, that could also be a solution (I don't know if/how that could be done).

**Comment by madoar** [comment]
This PR seems to have a conflict.

**Comment by plata** [comment]
Outdated.
